bricklayers.py

```diff
@@ -134,10 +135,38 @@ def process_gcode(input_file, layer_height, extrusion_multiplier):
     parser.add_argument("input_file", help="Path to the input G-code file")
     parser.add_argument("-layerHeight", type=float, default=0.2, help="Layer height in mm (default: 0.2mm)")
     parser.add_argument("-extrusionMultiplier", type=float, default=1, help="Extrusion multiplier for first layer (default: 1.5x)")
+    parser.add_argument("-bgcode", action='store_true', help="Input file is binary gcode (PrusaSlicer default from v2.7)")
     args = parser.parse_args()
 
+    bgcode = False
+    rename = False
+
+    input_path = Path(args.input_file)
+    if args.bgcode or input_path.suffix == '.bgcode':
+        # adjust file name to have proper suffix if necessary
+        if input_path.suffix != '.bgcode':
+            rename = True
+            input_path = input_path.rename(args.input_file + '.bgcode')
+        if os.system('bgcode ' + str(input_path)) == 0:
+            bgcode = True
+            # change name to reflect bgcode command's renaming of the file
+            input_path = input_path.parent / (input_path.stem + '.gcode')
+
     process_gcode(
-        input_file=args.input_file,
+        input_file=input_path,
         layer_height=args.layerHeight,
         extrusion_multiplier=args.extrusionMultiplier,
     )
+
+    # restore encoding if decoded
+    if bgcode:
+        os.system('bgcode ' + str(input_path))

```

Comment:
Prefer spawning with the `subprocess` module to `os.system` as the latter doesn't handle file names with spaces, quotes etc correctly. Something like (from my memory so could be wrong):

```python
subprocess.check_call(['bgcode', input_path])
```

Ideally though, it would be better if this `bgcode` command could output the changed file on stdout so we don't have to create so many temporary files.

Or even better, use a python library directly for parsing and writing bgcode.

Reply:
yes, the command's interface is not ideal for this use. my first thought was to use the python bindings, but pybgcode is not on pypi. and the bindings are not installed with libbgcode on gentoo. I did see a separate fedora package for pybgcode, but otherwise it seems like a more difficult install path.